subsys/bluetooth/controller/hci_driver.c

```diff
@@ -68,6 +68,12 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_CENTRAL) ||
 BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
 			 (CONFIG_BT_CTLR_SDC_PERIPHERAL_COUNT > 0));
 
+#if defined(CONFIG_BT_CTLR_CHANNEL_SOUNDING)
+BUILD_ASSERT(IS_ENABLED(CONFIG_BT_CTLR_SDC_CS_ROLE_INITIATOR) ||
```

Comment:
instead of this I suggest we use a `choice` kconfig https://docs.zephyrproject.org/latest/build/kconfig/setting.html#configuring-choices with 3 options:
* initiator only
* reflector only
* both (default)

this would enforce API sanity directly at the user config level